### PR TITLE
Update ZIO from 2.0.0-RC2 to 2.0.0-RC3

### PR DESCRIPTION
**Pull request by earldouglas**
This updates ZIO to 2.0.0-RC3, and replaces `ZManaged` with `Scope`.
For reference, see: <https://github.com/zio/zio/pull/6420>

### CODE REVIEW

File: zio-json/jvm/src/main/scala/zio/JsonPackagePlatformSpecific.scala [hunk 1]
```diff
@@ -9,22 +9,22 @@ import java.nio.charset.StandardCharsets
 import java.nio.file.{ Path, Paths }
 
 trait JsonPackagePlatformSpecific {
-  def readJsonAs(file: File): ZStream[Any, Throwable, ast.Json] =
+  def readJsonAs(file: File): ZStream[Scope, Throwable, ast.Json] =
```
Comment (jdegoes) [inline zio-json/jvm/src/main/scala/zio/JsonPackagePlatformSpecific.scala]: Ordinarily, streams bundle scopes within them, so there is no need to leak `Scope` on the outside. 

The only exception is if you want the resources consumed by the stream to stay open beyond the lifetime of the stream.

What do you think here and elsewhere?

File: zio-json/jvm/src/main/scala/zio/json/JsonDecoderPlatformSpecific.scala [hunk 2]
```diff
@@ -30,10 +30,10 @@ trait JsonDecoderPlatformSpecific[A] { self: JsonDecoder[A] =>
   final def decodeJsonStreamInput[R](
     stream: ZStream[R, Throwable, Byte],
     charset: Charset = StandardCharsets.UTF_8
-  ): ZIO[R, Throwable, A] =
+  ): ZIO[R with Scope, Throwable, A] =
     stream.toInputStream
-      .flatMap(is => ZManaged.fromAutoCloseable(UIO(new java.io.InputStreamReader(is, charset))))
-      .use(readAll)
+      .flatMap(is => ZIO.fromAutoCloseable(ZIO.succeed(new java.io.InputStreamReader(is, charset))))
```
Comment (jdegoes) [inline zio-json/jvm/src/main/scala/zio/json/JsonDecoderPlatformSpecific.scala]: Anytime you use `.use` in the old ZManaged API, it means you need to use `ZIO.scoped { ... }` in the new API, to close the open resources. Right now, they are left open when the function returns.

Comment (earldouglas) [reply zio-json/jvm/src/main/scala/zio/json/JsonDecoderPlatformSpecific.scala:35]: Would we just wrap the latter part?  e.g.:

```scala
    stream.toInputStream
      .flatMap(is =>
        ZIO.scoped {
          ZIO
            .fromAutoCloseable(ZIO.succeed(new java.io.InputStreamReader(is, charset)))
            .flatMap(readAll)
        }
```

File: zio-json/jvm/src/main/scala/zio/json/JsonDecoderPlatformSpecific.scala [hunk 3]
```diff
@@ -30,10 +30,15 @@ trait JsonDecoderPlatformSpecific[A] { self: JsonDecoder[A] =>
   final def decodeJsonStreamInput[R](
     stream: ZStream[R, Throwable, Byte],
     charset: Charset = StandardCharsets.UTF_8
-  ): ZIO[R, Throwable, A] =
-    stream.toInputStream
-      .flatMap(is => ZManaged.fromAutoCloseable(UIO(new java.io.InputStreamReader(is, charset))))
-      .use(readAll)
+  ): ZIO[R with Scope, Throwable, A] =
```
Comment (jdegoes) [inline zio-json/jvm/src/main/scala/zio/json/JsonDecoderPlatformSpecific.scala:33]: This `with Scope` needs to be deleted (!).

File: zio-json/jvm/src/main/scala/zio/json/JsonDecoderPlatformSpecific.scala [hunk 4]
```diff
@@ -43,21 +48,21 @@ trait JsonDecoderPlatformSpecific[A] { self: JsonDecoder[A] =>
    *
    * @see also [[decodeJsonStreamInput]]
    */
-  final def decodeJsonStream[R](stream: ZStream[R, Throwable, Char]): ZIO[R, Throwable, A] =
-    stream.toReader.use(readAll)
+  final def decodeJsonStream[R](stream: ZStream[R, Throwable, Char]): ZIO[R with Scope, Throwable, A] =
```
Comment (jdegoes) [inline zio-json/jvm/src/main/scala/zio/json/JsonDecoderPlatformSpecific.scala:51]: This `with Scope` needs to be deleted (!)

File: zio-json/jvm/src/main/scala/zio/json/JsonDecoderPlatformSpecific.scala (hunk 3 above)
Comment (earldouglas) [reply zio-json/jvm/src/main/scala/zio/json/JsonDecoderPlatformSpecific.scala:33]: See #601 for fix.

File: zio-json/jvm/src/main/scala/zio/json/JsonDecoderPlatformSpecific.scala (hunk 4 above)
Comment (earldouglas) [reply zio-json/jvm/src/main/scala/zio/json/JsonDecoderPlatformSpecific.scala:51]: See #601 for fix.

**Pull request merged**